packages/neon/neon_files/lib/widgets/browser_view.dart

```diff
@@ -47,9 +47,25 @@ class _FilesBrowserViewState extends State<FilesBrowserView> {
           stream: widget.bloc.path,
           builder: (final context, final pathSnapshot) => StreamBuilder<List<FilesTask>>(
             stream: widget.filesBloc.tasks,
-            builder: (final context, final tasksSnapshot) => !pathSnapshot.hasData || !tasksSnapshot.hasData
-                ? const SizedBox()
-                : BackButtonListener(
+            builder: (final context, final tasksSnapshot) {
+              if (!pathSnapshot.hasData || !tasksSnapshot.hasData) {
+                return const SizedBox();
+              }
+              return ValueListenableBuilder(
+                valueListenable: widget.bloc.options.showHiddenFilesOption,
+                builder: (final context, final showHiddenFiles, final _) {
+                  files.data?.removeWhere((final file) {
```

Comment:
This doesn't work correctly when you enable the option again because the hidden files were removed from the data.